pkgs/development/beam-modules/default.nix

```diff
@@ -23,19 +23,18 @@ let
         hexRegistrySnapshot = callPackage ./hex-registry-snapshot.nix { };
 
         rebar = callPackage ../tools/build-managers/rebar { };
-        rebar3-open = callPackage ../tools/build-managers/rebar3 {
-          hermeticRebar3 = false;
-        };
-        rebar3 = callPackage ../tools/build-managers/rebar3 {
-          hermeticRebar3 = true;
-        };
+        rebar3-open = callPackage ../tools/build-managers/rebar3 { };
+        rebar3 = callPackage ../tools/build-managers/rebar3 { };
```

Comment:
These need the `hermeticRebar3` option. It defaults to true, which means `rebar3-open` and `rebar3` are now the same things. And users of `rebar3-open` will be unable to download packages from hex because it is hermetic.

Reply:
After going through it a little further, I see that you removed the hermetic patch, which means (I think) it now goes against how Nix is expecting things. It is free to download anything that it wants instead of you explicitly saying what the applications dependencies are. I would like someone with more knowledge than myself to say whether this is acceptable or not.

Reply:
Hi,

In short, removal of the hermetic patch isn't against Nix's way of doing things: nix-build runs with network isolation (unless it's a fixed-output derivation) therefore rebar3 won't be able to download anything during the process of build.

Fetching dependencies is a different story. It was moved to a separate step. Using `rebar3 get-deps` produces fixed output (rebar3 itself uses sha256 to freeze the dependencies), so this is no worse than using `fetchurl`.

I described my reasoning more in depth here: https://github.com/NixOS/nixpkgs/issues/53834